test/unit/default_db_attr_tests.rb

```diff
@@ -0,0 +1,166 @@
+require 'assert'
+require 'ardb/default_db_attr'
+
+require 'ardb/record_spy'
+
+module Ardb::DefaultDbAttr
+
+  class UnitTests < Assert::Context
+    desc "Ardb::DefaultDbAttr"
+    setup do
+      @record_class = Class.new do
+        include DefaultDbAttrRecordSpy
+        include Ardb::DefaultDbAttr
+      end
+    end
+    subject{ @record_class }
+
+    should have_imeths :default_db_attr, :default_db_attributes
+
+    should "know its default db attributes" do
+      assert_equal [], subject.default_db_attributes
+    end
+
+    should "add default db attributes using `default_db_attr`" do
+      attr_name = Factory.string
+      subject.default_db_attr(attr_name)
+      assert_includes attr_name, subject.default_db_attributes
+
+      attr_names = [Factory.string, Factory.string.to_sym]
+      subject.default_db_attr(*attr_names)
+      attr_names.each do |attr_name|
+        assert_includes attr_name.to_s, subject.default_db_attributes
+      end
+    end
+
+    should "not add duplicate db attributes using `default_db_attr`" do
+      attr_name = Factory.string
+      subject.default_db_attr(attr_name)
+      assert_equal [attr_name], subject.default_db_attributes
+
+      subject.default_db_attr(attr_name)
+      assert_equal [attr_name], subject.default_db_attributes
+
+      more_attr_names = [attr_name, Factory.string]
+      subject.default_db_attr(*more_attr_names)
+      exp = ([attr_name] + more_attr_names).uniq
+      assert_equal exp, subject.default_db_attributes
+    end
+
+    should "add an around create callback" do
+      callback = subject.callbacks.first
+      assert_equal :around_create, callback.type
+      exp = [:allow_db_to_default_attrs_for_default_db_attr]
+      assert_equal exp, callback.args
+    end
+
+  end
+
+  class InitTests < UnitTests
+    desc "when init"
+    setup do
+      @db_attrs = Factory.integer(3).times.map{ Factory.string }
+      @record_class.default_db_attr(*@db_attrs)
+
+      @record = @record_class.new
+    end
+    subject{ @record }
+
+  end
+
+  class AroundCreateCallbackTests < InitTests
```

Comment:
@jcredding let's just put this all in the `InitTests` context since there are no other contexts right now.  I think it reads better.  You could also move the around callback setup into a private method for organization if you want.